src/models/thread.ts

```diff
@@ -504,17 +504,71 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
         throw new Error("Unsupported function on the thread model");
     }
 
+    /**
+     * Get the ID of the event that a given user has read up to, or null if we
+     * have received no read receipts from them.
+     * @param userId - The user ID to get read receipt event ID for
+     * @param ignoreSynthesized - If true, return only receipts that have been
+     *                            sent by the server, not implicit ones generated
+     *                            by the JS SDK.
+     * @returns ID of the latest event that the given user has read, or null.
+     */
+    public getEventReadUpTo(userId: string, ignoreSynthesized?: boolean): string | null {
+        const isCurrentUser = userId === this.client.getUserId();
+        const lastReply = this.lastReply();
+        if (isCurrentUser && lastReply) {
+            // If the last activity in a thread is prior to the first threaded read receipt
+            // sent in the room (suggesting that it was sent before the user started
+            // using a client that supported threaded read receipts), we want to
+            // consider this thread as read.
+            const beforeFirstThreadedReceipt = (lastReply.getTs() ?? 0) < this.room.getOldestThreadedReceiptTs();
+            if (beforeFirstThreadedReceipt) {
+                return this.timeline.at(-1)?.getId() ?? null;
```

Comment:
is it even possible for `this.timeline.at(-1)` to be nullish here, given lastReply is truthy? If not, let's avoid the `?? null` please.

Reply:
`getId()` can be `undefined`

Reply:
Hrm, I think it should only be `undefined` for unsent events, which should not be relevant here?

In any case we can still assume that `this.timeline.at(-1)` is non-null?

If you want to keep the `?? null` please can you document it in the doc-comment: ie, that this function can return null under particular circumstances (and what they are)? It's super-frustrating to realise that a function sometimes return null and then have to spend hours reverse-engineering when it happens and whether we need to care about it.

Reply:
I've updated the logic slightly to avoid the optional chaining here